### ISSUE TITLE
[Playbook] Playbooks apply status to entities that shouldn't have them

### ISSUE THREAD

**Issue by Lhorus6** [LuatixHQ/opencti#9071]
## Description

When I change statuses on entities with my playbook, it adds the status on all entities in the manipulated bundle, without taking into account which ones should have it.

## Environment

OCTI 6.2.12

## Reproducible Steps

Steps to create the smallest reproducible scenario:
1. Create a playbook:
2. 1st component: listening on "create" + "Entity type=Note"
3. 2nd component: Apply predefined rule - Resolve container references
4. 3rd component: Manipulate knowledge - "Replace > Status > [an Incident status]", with "Manipulate all elements included in the bundle" turn on
5. 4th component: Send for ingestion
 
![Image](https://github.com/user-attachments/assets/867bd0e9-08e3-40d8-9d70-e6f11bd8c298)

6. Go on an Indicent
7. Create a Note

## Expected Output

I want the status to be changed on my Incident **only**

## Actual Output

My status is changed on my Incident, but it is also applied to my Note (because it is included in the bundle manipulated by the playbook). However:

1. The Note has no workflow (i.e. the "processing status" field is normally "disable"), so should not have a status.
2. The status applied is an Incident status, it should not apply to entity types other than Incident.

![Image](https://github.com/user-attachments/assets/942f788b-b6ee-436e-88e7-f0ba16abfe25)

![Image](https://github.com/user-attachments/assets/9dfcc59a-a9fc-4980-b10a-3c9c46c9870e)

**Comment by romain-filigran** [comment]
I agree. It is not normal to apply a status that is not defined in the entity status workflow.

**Comment by JeremyCloarec** [comment]
Here are my findings after reproducing the bug and trying stuff out with updating statuses on entities:
* there is no safety check on statuses when field patching: any status can be added on an entity, even if the entity doesn't have any workflow enabled. This can be fixed by verifying the status, and throwing an error if the status isn't valid for the current entity
* what should be the behavior when upserting ? should we keep other changes and simply ignore the status if it is incorrect? or should we throw an error?
* I'm not sure what to do about playbooks: should we add a new feature to be able to apply a filter on the manipulate knowledge component @nino-filigran  @romain-filigran ? For now, since all entities can have a status, the playbook cannot manipulate only the entity that the status applies to

**Comment by nino-filigran** [comment]
- there is no safety check on statuses when field patching: any status can be added on an entity, even if the entity doesn't have any workflow enabled. This can be fixed by verifying the status, and throwing an error if the status isn't valid for the current entity
   - Let's not add a check: simply, the status should not be apply if not part of the statuses of the entity 
- what should be the behavior when upserting ? should we keep other changes and simply ignore the status if it is incorrect? or should we throw an error?
   - We should simply ignore
- I'm not sure what to do about playbooks: should we add a new feature to be able to apply a filter on the manipulate knowledge component @nino-filigran @romain-filigran ? For now, since all entities can have a status, the playbook cannot manipulate only the entity that the status applies to
   - like response 1 if possible :)
